staging/src/k8s.io/kubectl/pkg/cmd/profiling.go

```diff
@@ -30,12 +30,12 @@ var (
 	profileOutput string
 )
 
-func addProfilingFlags(flags *pflag.FlagSet) {
+func AddProfilingFlags(flags *pflag.FlagSet) {
 	flags.StringVar(&profileName, "profile", "none", "Name of profile to capture. One of (none|cpu|heap|goroutine|threadcreate|block|mutex)")
 	flags.StringVar(&profileOutput, "profile-output", "profile.pprof", "Name of the file to write the profile to")
 }
 
-func initProfiling() error {
+func InitProfiling() error {
```

Comment:
not sure how I feel about exporting this modifying global package vars. would something like this be better?

```go
type ProfileOptions struct {
  ProfileName   string
  ProfileOutput string
}
func (p *ProfileOptions) AddFlags(flags *pflag.FlagSet) {
  ... bind to p.ProfileName, p.ProfileOutput ...
}
func (p *ProfileOptions) Init() error {
  ... switch on p.ProfileName ...
}
func (p *ProfileOptions) func Flush() error {
  ... switch on p.ProfileName, flush to p.ProfileOutput ...
}

Reply:
and that options struct could be under `k8s.io/kubectl/pkg/util/profiling`

Reply:
I'll take this as a follow up since this PR is a rebase magnet

Reply:
I'll follow up on this in the next PR if that is OK.